style.css

```diff
@@ -363,8 +363,18 @@ label[for="nav_toggle"] { /* The MENU button */
 .fn {
     transition: background-color 100ms ease-in-out;
 }
-.fn:hover {
-    background-color: hsla(calc(75 - var(--fn) * 10), 80%, calc(100% - 3%*var(--fn)));
+
+.def .fn:hover {
+    background-color: hsla(calc(187 - (var(--fn) - 1) * 10), 61%, calc(100% - 3%*var(--fn)));
+}
+.theorem .fn:hover {
+    background-color: hsla(calc(115 - (var(--fn) - 1) * 10), 62%, calc(100% - 3%*var(--fn)));
+}
+.axiom .fn:hover, .constant .fn:hover {
+    background-color: hsla(calc(16 - (var(--fn) - 1) * 10), 94%, calc(100% - 3%*var(--fn)));
+}
+.structure .fn:hover, .inductive .fn:hover {
+    background-color: hsla(calc(40 - (var(--fn) - 1) * 10), 98%, calc(100% - 3%*var(--fn)));
 }
```

Comment:
I wonder if this would look better without the hue variance,
```suggestion
.def .fn:hover {
    background-color: hsla(calc(187, 61%, calc(100% - 3%*var(--fn)));
}
.theorem .fn:hover {
    background-color: hsla(calc(115, 62%, calc(100% - 3%*var(--fn)));
}
.axiom .fn:hover, .constant .fn:hover {
    background-color: hsla(calc(16, 94%, calc(100% - 3%*var(--fn)));
}
.structure .fn:hover, .inductive .fn:hover {
    background-color: hsla(calc(40, 98%, calc(100% - 3%*var(--fn)));
}
```

Reply:
![image](https://user-images.githubusercontent.com/425260/94141998-7c5f6500-fe65-11ea-9fa5-2af478b68e68.png)

Reply:
I was skeptical about this since I thought it would blend together too much, but actually, with the dynamics of the colors darkening as you move your mouse, I think this is the best solution. I even prefer it with 4% steps instead of 5%.

Reply:
It would be cool if we could count up in luminosity from the hovered expression, instead of counting down from the ancestor. I can't think of a way to do that with just CSS though.